types/block.go

```diff
@@ -268,6 +268,7 @@ func (b *Block) fillDataAvailabilityHeader() {
 
 	// return the root hash of DA Header
 	b.DataHash = b.DataAvailabilityHeader.Hash()
+	b.NumOriginalDataShares = int64(dataSharesLen)
```

Comment:
I might be wrong but what is returned here:
https://github.com/lazyledger/lazyledger-core/blob/ece16550a1faa124450001b77eb340cd4424374a/types/block.go#L1398
isn't exactly what the spec specifies as 
https://github.com/lazyledger/lazyledger-core/blob/ece16550a1faa124450001b77eb340cd4424374a/types/block.go#L1390-L1398
will also include the tail padding shares.
So I think `ComputeShares` should return `curLen` instead: 
https://github.com/lazyledger/lazyledger-core/blob/ece16550a1faa124450001b77eb340cd4424374a/types/block.go#L1386
(cc @adlerjohn)